Automattic/woocommerce-payments
# Limit product_code in Level3 data shipping-only order within 12 characters

Fixes #4738

#### Changes proposed in this Pull Request

The PR prevents the error seen for WP Admin-initiated order payments, with only shipping charges added, without products or any other custom fees in the order. Currently, there is an error seen in the payment intent log similar to the one below:

```
400 ERR A request to update a PaymentIntent pi_3LerMv2EhQJVb2mF0FpXTsq6 failed
\"error\": {

    \"message\": \"Invalid string: zero-cost-...o-cost-fee; must be at most 12 characters\",

    \"param\": \"level3[line_items][0][product_code]\",
```

* The length of the existing `product_code` string  sent within Level 3 data to stripe  is 13 characters - `zero-cost-fee` - one more than [the limit of 12](https://stripe.com/docs/level3#:~:text=product_code:%20Up%20to%2012%20characters%20that%20uniquely%20identify%20the%20product.). ( Documentation link visible when you are logged in to Stripe dashboard ) 

* The PR changes `product_code` from `zero-cost-fee` to `no-cost-fee` to bring it below 12 characters and avoid the error. 

#### Testing instructions
1. Create a new order from `WP admin > WooCommerce > Orders > New order`
2. Do **not** add any products. 
3. Click Add Items, then Add Shipping to add only a shipping fee to the product. Add a non-zero shipping fee to the order. 
4. Complete the order and check transaction log within `Status > Logs` 
5. You should no longer see the `invalid string` error, and should see the `product_code` set as `no-cost-fee`

-------------------


**Post merge**
